tools/single_nuc.py

```diff
@@ -0,0 +1,669 @@
+import matplotlib.pyplot as plt
+import pandas as pd
+import numpy as np
+import re
+
+#-------------------- Helper Functions --------------------
+
+def select_sublist(x, y):
+    '''
+    Select intersection of list x and list y. Additionally, capture indices
+        of said intersection on list x.
+
+    Arguments:
+        x (list): Reference list.
+        y (list): Smaller list sharing some or all elements with x.
+
+    Returns:
+        sublist (list): Intersection of x and y.
+        indices (list): Index or indices of intersection on x.
+    '''
+
+
+    sublist = [
+        val for i, val in enumerate(x, start=1)
+        if val in y or i in (y if isinstance(y, list) else [y])
+        ]
+    indices = [i for i, val in enumerate(x) if val in sublist]
+    return sublist, indices
+
+def process_metadata(
+        gen_csv='',
+        existing_df=None,
+        inp_datalibs=[],
+        inp_variables=[],
+        inp_units=[]):
+    
+    datalibs = ['fendl2', 'fendl3']
+    variables = [
+        'Number_Density', 'Specific_Activity',
+        'Total_Decay_Heat', 'Contact_Dose'
+        ]
+    units = ['atoms_kg', 'Bq_kg', 'W_kg', 'Sv_hr_._data_ANS6_4_3']
+
+    if inp_datalibs:
+        datalibs, _ = select_sublist(datalibs, inp_datalibs)
```

Comment:
You might not need the `select_sublist()` method if these are replaced by a single line if using set intersection (see above)